index.bs

```diff
@@ -2166,71 +2428,121 @@ interface MLGraph {
         The underlying implementation provided by the User Agent.
 </dl>
 
-<dl dfn-type=method dfn-for=MLGraph>
-    : <dfn>compute(inputs, outputs)</dfn>
+## MLCommandEncoder ## {#api-mlcommandencoder}
+The {{MLCommandEncoder}} interface represents a method of execution that synchronously records the computational workload of a compiled graph {{MLGraph}} to a GPU command buffer {{GPUCommandBuffer}} on the calling thread. Since the workload is not immediately executed, just recorded, this method allows more flexibility for the caller to determine how and when the recorded commands will be submitted for execution on the GPU relative to other GPU workload on the same queue.
+
+<script type=idl>
+typedef (GPUBuffer or GPUTexture) MLGPUResource;
+
+dictionary MLGPUInput {
+  required MLGPUResource resource;
+  required sequence<long> dimensions;
+};
+
+typedef record<DOMString, (MLGPUResource or MLGPUInput)> MLNamedGPUInputs;
+typedef record<DOMString, MLGPUResource> MLNamedGPUOutputs;
+
+[SecureContext, Exposed=(Window, DedicatedWorker)]
+interface MLCommandEncoder {
+  undefined initializeGraph(MLGraph graph, MLNamedGPUInputs inputs);
```

Comment:
We may want to name `inputs` arguments passed to `initializeGraph()` and `dispatch()` differently to make it clearer their semantics differ. I think this would improve developer ergonomics.

Reply:
I will add more descriptive text into describing these two methods in more detail. I dont think just changing the param name will make any material difference.

Reply:
BTW, I'll encourage other reviewers to take a look at this PR and given their opinion on it relative to #255. I hope we can come to an agreement on this important technical topic soon. It has been a while in the making.

Reply:
Please look at the latest commit. I've removed `MLCommandEncoder` and let `computeAsync` call handle both the CPU and GPU async use case per discussion with @RafaelCintron and @bbernhar. Detail please refer to this reply https://github.com/webmachinelearning/webnn/pull/255#issuecomment-1079981083